simulator/simulator.js

```diff
@@ -199,8 +200,14 @@ prosumerBuffer = async function () {
                 let conserve = -diff * (1 - prosumer.under_production_buy)
 
                 if(prosumer.balance >= buy){
-                    await server_db_utils.updateBalanceById(prosumer.id, (prosumer.balance - buy).toFixed(2))
-                    await server_db_utils.updateBufferById(prosumer.id, (prosumer.buffer - conserve).toFixed(2))
+                    if(prosumer.buffer - buy < 0){
+                        await server_db_utils.updateBalanceById(prosumer.id, (prosumer.balance - buy).toFixed(2))
```

Comment:
Maybe I'm a bit lost... but can't the balance end up negative here?

Reply:
Hm... I'm very confused :smile: We are comparing the `buy` variable to the `prosumer.buffer` and then subtracting with `prosumer.balance`. What is `buy`?

Reply:
I know, this buffer part is quite confusing to say at least. 

Basically, buy is how much electricity you bought times the current electricity price. The balance is updated with this variable.

The buffer is not updated with the buy variable and you are right, it should be replaced with the conserve variable because it is not based on the price.

Reply:
I resolved it by changing the variable of the condition to conserve, because that is what updates the buffer. Now the buffer may not become negative nor will the balance. 

```suggestion
                        if(prosumer.buffer - conserve< 0){
```